[udp] Updated sample to use the newer chrome.sockets.udp API

lostsource opened the pull request:
Thought this sample should be updated since chrome.socket API has been deprecated for quite a while.

beaufortfrancois commented:
Will have a look at it Tuesday for sure ;)

beaufortfrancois commented on samples/udp/networking.js:
```diff
@@ -4,54 +4,47 @@
-              }
-            };
-            self.poll();
+              }              
```
Do you mind removing trailing whitespaces there?

beaufortfrancois commented on samples/udp/networking.js:
```diff
@@ -4,54 +4,47 @@
             connected = (connectResult == 0);
-            socket.ondata = function(result) {
+
+            chrome.sockets.udp.onReceive.addListener(function(result){
```
`s/chrome.sockets.udp/socket/`

beaufortfrancois commented on samples/udp/networking.js:
```diff
@@ -107,7 +102,7 @@
       // Clients
       clients: {
         udp: {
```
Since we don't use other types of clients, we may want to remove the "udp" key and simply call `var ec = new chromeNetworking.clients.echoClient();` in demo.js. What do you think?

beaufortfrancois commented:
Thanks Joseph! Small nits there!
Moreover as in https://github.com/GoogleChrome/chrome-app-samples/pull/385, can you add a space between function params?

lostsource commented:
Hi Francois, 

I updated this branch as per your comments above.. however I'm not sure if I made the right thing here. Basically I re-based and squashed several commits into commit e3d1fda and push (forced) to update this PR.

Would this be the right approach to squash the 14 commits in PR 385 to a single commit ?

beaufortfrancois commented:
That's perfect Joseph!
Thank you really much for your work there!
